front-end/src/screens/SignupForm/index.tsx

```diff
@@ -65,7 +65,7 @@ const SignupForm = ({ className }:Props): JSX.Element => {
 								className={errors.username ? 'error' : ''}
 								name='username'
 								placeholder='john'
-								ref={register({ minLength: 3, required: true })}
+								ref={register({ pattern: /^[A-Za-z0-9._-]{3,20}$/, required: true })}
```

Comment:
like here, it'd be cool to have something like:
```suggestion
								ref={register({ maxLength:20, minLength:3, pattern: /^[A-Za-z0-9._-]*$/, required: true })}
```
So that we can have specific errors like:
```js
{errors.username?.type === 'required' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_REQUIRED_ERROR}</span>}
{errors.username?.type === 'minLength' && <span className={'errorText'}>{messages.VALIDATION_USERNAME_MINLENGTH_ERROR}</span>} ...
```
I've had this bloody type error though, not alone here: https://github.com/react-hook-form/react-hook-form/issues/728
You can have a brief look, if you manage to solve that. If not it's ok, we have more important things to solve.

Reply:
done

Reply:
fixed with type assertion